File: CS/E5023/Models/DataHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data;
using System.Data.Odbc;

namespace E5023.Models {
    public class DataHelper {
        static OdbcConnection GetConnection() {
            try {
                OdbcConnection connection = new OdbcConnection("Driver={Microsoft Access Driver (*.mdb)};Dbq=|DataDirectory|\\nwind.mdb;Uid=;Pwd=;");
                connection.Open();
                return connection;
            }
            catch (Exception ex) {
                throw new Exception(String.Format("Connection failed!: {0}", ex.Message), ex);
            }
        }

        public static object ExecuteCommand(string commandFormat, params object[] commandParameters) {
            try {
                string commandText = String.Format(commandFormat, commandParameters);
                using (OdbcConnection connection = GetConnection()) {
                    OdbcCommand command = new OdbcCommand(commandText, connection);
                    object result = command.ExecuteScalar();
                    connection.Close();
                    return result;
                }
            }
            catch (Exception ex) {
                throw new Exception(String.Format("Command executing failed!: {0}", ex.Message), ex);
            }
        }

        public static DataTable ProcessSelectCommand(string commandFormat, params object[] commandParameters) {
            try {
                string commandText = String.Format(commandFormat, commandParameters);
                using (OdbcConnection connection = GetConnection()) {
                    OdbcDataAdapter adapter = new OdbcDataAdapter(commandText, connection);
                    DataTable table = new DataTable();
                    adapter.Fill(table);
                    return table;
                }
            }
            catch (Exception ex) {
                throw new Exception(String.Format("Data selecting failed!: {0}", ex.Message), ex);
            }
        }
    }
}

File: VB/E5023/Models/DataHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Data
Imports System.Data.Odbc

Namespace E5023.Models
	Public Class DataHelper
		Private Shared Function GetConnection() As OdbcConnection
			Try
				Dim connection As New OdbcConnection("Driver={Microsoft Access Driver (*.mdb)};Dbq=|DataDirectory|\nwind.mdb;Uid=;Pwd=;")
				connection.Open()
				Return connection
			Catch ex As Exception
				Throw New Exception(String.Format("Connection failed!: {0}", ex.Message), ex)
			End Try
		End Function

		Public Shared Function ExecuteCommand(ByVal commandFormat As String, ParamArray ByVal commandParameters() As Object) As Object
			Try
				Dim commandText As String = String.Format(commandFormat, commandParameters)
				Using connection As OdbcConnection = GetConnection()
					Dim command As New OdbcCommand(commandText, connection)
					Dim result As Object = command.ExecuteScalar()
					connection.Close()
					Return result
				End Using
			Catch ex As Exception
				Throw New Exception(String.Format("Command executing failed!: {0}", ex.Message), ex)
			End Try
		End Function

		Public Shared Function ProcessSelectCommand(ByVal commandFormat As String, ParamArray ByVal commandParameters() As Object) As DataTable
			Try
				Dim commandText As String = String.Format(commandFormat, commandParameters)
				Using connection As OdbcConnection = GetConnection()
					Dim adapter As New OdbcDataAdapter(commandText, connection)
					Dim table As New DataTable()
					adapter.Fill(table)
					Return table
				End Using
			Catch ex As Exception
				Throw New Exception(String.Format("Data selecting failed!: {0}", ex.Message), ex)
			End Try
		End Function
	End Class
End Namespace